components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java

```diff
@@ -139,11 +138,35 @@ public boolean checkSelfRegistrationShowUsernameUnavailability(String tenant)
      *
      * @param tenant tenant domain name.
      * @return returns true if  username recovery enabled.
-     * @throws PreferenceRetrievalClientException
+     * @throws PreferenceRetrievalClientException If any PreferenceRetrievalClientException occurs.
      */
     public boolean checkUsernameRecovery(String tenant) throws PreferenceRetrievalClientException {
 
-        return checkPreference(tenant, RECOVERY_CONNECTOR, USERNAME_RECOVERY_PROPERTY);
+        return checkPreference(tenant, RECOVERY_CONNECTOR, IdPManagementConstants.USERNAME_RECOVERY_PROPERTY);
+    }
+
+    /**
+     * Check email based username recovery is enabled or not.
+     *
+     * @param tenant tenant domain name.
+     * @return returns true if email based username recovery enabled.
+     * @throws PreferenceRetrievalClientException If any PreferenceRetrievalClientException occurs.
+     */
+    public boolean checkEmailBasedUsernameRecovery(String tenant) throws PreferenceRetrievalClientException {
```

Comment:
Do we need to have two separate methods for this `checkEmailBasedUsernameRecovery` and `checkSMSBasedUsernameRecovery`? Instead can't we modify the method signature to get the recovery property?

Reply:
Yes we can. But since all other retrieval functions (password recovery config retrievals) follows this pattern I have opted to have separate functions for each config.

Reply:
We can merge the multiple methods to one.

Reply:
To merge the functions, still we need to pass the config value name as a parameter to that single function. Then we have to retrieve that constant in the FE and send it to BE again. The current implementation don't have that complexity.